### DISCUSSION TITLE
Is there a way to disallow drag handle to drop nodes into other nodes?

### DISCUSSION THREAD

**Discussion by jofri** [ueberdosis/tiptap#5622] [General Questions]
I'm using [Tiptap drag-handle](https://tiptap.dev/docs/editor/extensions/functionality/drag-handle-react) in my Next.js app to allow users to move around block quotes in the editor. I'm calling block quotes "scenes" in my editor.

However, moving block quotes around has two issues:

1. When dropping a block quote outside of another block quote (in the doc), the block quote disappears and the node becomes a paragraph.  

2. I don't want to allow the block quote to be dragged into another block quote / list or other node - only allow them to be rearranged on the top level.

Is there any way to obtain this behaviour on any pointes for how to configure the drag handle extension?

<img width="320" alt="Screenshot 2024-09-16 at 22 21 47" src="https://github.com/user-attachments/assets/3cf2e396-6db4-4ff0-b84d-53005f359d52">

Kind regards,

**Comment by GhostyCHEN** [comment]
only allow them to be rearranged on the top level! Me too

**Comment by guanriyue** [comment]
ProseMirror's `dropcursor` only checks whether the current position allows the dragged content to be placed, and this judgment comes from your schema definition. If allowed, a horizontal line will appear as a marker.  

Refer to ProseMirror's `dropcursor` to implement your own custom `dropcursor`. This part requires you to handle it yourself.

**Comment by OlenaPodranetska** [comment]
An example for handling the drop action (checking whether the current position allows the dragged content to be placed, thanks @guanriyue for the hint):
```
handleDrop(view, event, slice, moved) {
	const maxListLevel = 3;
	const coords = { left: event.clientX, top: event.clientY };
	const pos = view.posAtCoords(coords)?.pos;

	if (pos == null) return false;

	const $pos = view.state.doc.resolve(pos);

	// Your conditions here
	const currentLevel = getListNestingLevel($pos);
	if (currentLevel >= maxListLevel) {
		// Deny the drop action
		return true;
	}

	if (moved) {
		view.dispatch(view.state.tr.setMeta('drop', true));
	}

	return false; // Allow drop
},
```
